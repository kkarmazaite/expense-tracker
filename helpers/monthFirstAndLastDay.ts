export const monthFirstAndLastDay = (date:Date) => {
  const firstDay = new Date(date.getFullYear(), date.getMonth(), 1);
  const lastDay = new Date(date.getFullYear(), date.getMonth() + 1, 0);

  return {
    firstDay,
    lastDay,
  }
}