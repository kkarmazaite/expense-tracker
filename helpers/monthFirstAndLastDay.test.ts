import { expect, describe, it } from 'vitest'
import { monthFirstAndLastDay } from './monthFirstAndLastDay'

describe('Helper | monthFirstAndLastDay', async() => {
  it('Check monthFirstAndLastDay function', () => {
    const input = new Date('2023-04-19')
    const output = {
      firstDay:new Date('2023-04-01'),
      lastDay: new Date('2023-04-30'),
    }

    expect(monthFirstAndLastDay(input)).toStrictEqual(output)
    expect(monthFirstAndLastDay(input)).toMatchSnapshot()
  })
})
