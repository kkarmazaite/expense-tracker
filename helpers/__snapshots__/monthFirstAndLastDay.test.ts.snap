// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Helper | monthFirstAndLastDay > Check monthFirstAndLastDay function 1`] = `
{
  "firstDay": 2023-04-01T00:00:00.000Z,
  "lastDay": 2023-04-30T00:00:00.000Z,
}
`;
